### PR TITLE
Adding provisioning and server db config docs

### PR DESCRIPTION
**Pull request by kdhamric**
Adding provisioning and server db config docs. Did not add an overall 'what is resources' as our cli is fairly fragmented right now. We need to add a resources page once we put out the CLI improvements.

### CODE REVIEW

File: docs/docs/configuration/provisioning.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Provisioning server
+
+Tracetest allows a server to be provisioned the first time a new Tracetest server is installed and launched. Provisioning sets certain resources in the server to the specified values, allowing you to configure the server. It is convenient in a CI/CD flow where you want to launch a server with a specified configuration.
```
Comment (jfermi) [inline docs/docs/configuration/provisioning.md:3]:
```suggestion
Tracetest allows a server to be provisioned the first time Tracetest is installed and launched. Provisioning sets certain resources in the server to the specified values, allowing you to configure the server. It is convenient in a CI/CD flow where you want to launch a server with a specified configuration. 
```

File: docs/docs/configuration/provisioning.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# Provisioning server
+
+Tracetest allows a server to be provisioned the first time a new Tracetest server is installed and launched. Provisioning sets certain resources in the server to the specified values, allowing you to configure the server. It is convenient in a CI/CD flow where you want to launch a server with a specified configuration. 
+
+The server is provisioned by specifying a series of yaml snippets which will configure various resources. Each snippet is separated with the yaml separator, ---.
```
Comment (jfermi) [inline docs/docs/configuration/provisioning.md:5]:
```suggestion
The server is provisioned by specifying a series of YAML snippets which will configure various resources. Each snippet is separated with the YAML separator, ---.
```

File: docs/docs/configuration/provisioning.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+Currently, the following resources can be provisioned: 
+- DataStore
+- PollingProfile
+- Config
+- Demo
+
+For docker based installs, the provisioning file is placed in the ./tracetest/tracetest-provisioning.yaml file by default when you run the 'tracetest server install' command and select the 'Using Docker Compose' option. The first time you start tracetest with a 'docker compose -f tracetest/docker-compose.yaml  up -d' command, the server will use the contents of this file to provision the server. To provision differently, you would alter the contents of the tracetest-provisioning.yaml file before launching tracetest in docker.
```
Comment (jfermi) [inline docs/docs/configuration/provisioning.md:13]:
```suggestion
For Docker based installs, the provisioning file is placed in the ./tracetest/tracetest-provisioning.yaml file by default when you run the 'tracetest server install' command and select the 'Using Docker Compose' option. The first time you start Tracetest with a 'docker compose -f tracetest/docker-compose.yaml  up -d' command, the server will use the contents of this file to provision the server. To provision differently, you would alter the contents of the tracetest-provisioning.yaml file before launching Tracetest in Docker.
```

File: docs/docs/configuration/provisioning.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+spec:
+  name: pokeshop
+  type: pokeshop
+  enabled: true
+  pokeshop:
+    httpEndpoint: http://demo-api:8081
+    grpcEndpoint: demo-api:8082
+```
+
+Alternatively, we support setting an environment variable called TRACETEST_PROVISIONING to provision the server when it first is started. Take the provisioning YAML you want to utilize, base64 encode it, and set the TRACETEST_PROVISIONING environment variable with the result. The Tracetest server will then provision based on the base64 encoded provisioning data in this environment variable the first time it is launched.
```
Comment (jfermi) [inline docs/docs/configuration/provisioning.md:49]:
```suggestion
Alternatively, we support setting an environment variable called TRACETEST_PROVISIONING to provision the server when it is first started. Base64 encode the provisioning YAML you want to utilize and set the TRACETEST_PROVISIONING environment variable with the result. The Tracetest server will then provision based on the Base64 encoded provisioning data in this environment variable the first time it is launched.
```

File: docs/docs/configuration/server.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# Configuring the Tracetest server
```
Comment (jfermi) [inline docs/docs/configuration/server.md:1]:
```suggestion
# Configuring the Tracetest Server
```

File: docs/docs/configuration/server.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+Tracetest requires a very minimal configuration to be launched, needing just the connection information to connect with the PostgreSQL database which is installed as part of the server install. There are a couple ways to provide this database connection information.
```
Comment (jfermi) [inline docs/docs/configuration/server.md:3]:
```suggestion
Tracetest requires a very minimal configuration to be launched, needing just the connection information to connect with the PostgreSQL database which is installed as part of the server install. There are a couple of ways to provide this database connection information.
```

File: docs/docs/configuration/server.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+For docker based installs, the server configuration file is placed in the ./tracetest/tracetest.yaml file by default when you run the 'tracetest server install' command and select the 'Using Docker Compose' option. The configuration file is mounted to `/app/config.yaml` within the Tracetest Docker container. When tracetest is run with a 'docker compose -f tracetest/docker-compose.yaml  up -d' command, the server will use the contents of this file to connect to the Postgres database. All other configuration data is stored in the Postgres instance.
```
Comment (jfermi) [inline docs/docs/configuration/server.md:5]:
```suggestion
For Docker-based installs, the server configuration file is placed in the ./tracetest/tracetest.yaml file by default when you run the 'tracetest server install' command and select the 'Using Docker Compose' option. The configuration file is mounted to `/app/config.yaml` within the Tracetest Docker container. When Tracetest is run with a 'docker compose -f tracetest/docker-compose.yaml  up -d' command, the server will use the contents of this file to connect to the Postgres database. All other configuration data is stored in the Postgres instance.
```

File: docs/docs/configuration/server.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+postgres:
+  host: postgres
+  user: postgres
+  password: postgres
+  port: 5432
+  dbname: postgres
+  params: sslmode=disable
+```
+
+Alternatively, we support setting ana series of environment variables which can contain the connection information for the Postgres instance. If these environment variables are set, they will be used by the Tracetest server to connect to the database. The variables are:
```
Comment (jfermi) [inline docs/docs/configuration/server.md:19]:
```suggestion
Alternatively, we support setting a series of environment variables which can contain the connection information for the Postgres instance. If these environment variables are set, they will be used by the Tracetest server to connect to the database.
```

File: docs/docs/configuration/server.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+The list of environment variables, and example values is:
```
Comment (jfermi) [inline docs/docs/configuration/server.md:21]:
```suggestion
The list of environment variables and example values is:
```

**Review by jfermi — COMMENTED** [review]
Some editing suggestions.

File: docs/docs/configuration/provisioning.md (hunk 1 above)
Comment (kdhamric) [reply docs/docs/configuration/provisioning.md:3]: I think I may want to leave this wording to distinguish between a server install and the CLI install. Maybe it is not needed???

File: docs/docs/configuration/server.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
+# Configuring the Tracetest Server
+
+Tracetest requires a very minimal configuration to be launched, needing just the connection information to connect with the PostgreSQL database which is installed as part of the server install. There are a couple of ways to provide this database connection information.
+
+For Docker-based installs, the server configuration file is placed in the ./tracetest/tracetest.yaml file by default when you run the 'tracetest server install' command and select the 'Using Docker Compose' option. The configuration file is mounted to `/app/config.yaml` within the Tracetest Docker container. When Tracetest is run with a 'docker compose -f tracetest/docker-compose.yaml  up -d' command, the server will use the contents of this file to connect to the Postgres database. All other configuration data is stored in the Postgres instance.
```
Comment (adnanrahic) [inline docs/docs/configuration/server.md:5]:
```suggestion
For Docker-based installs, the server configuration file is placed in the `./tracetest/tracetest.yaml` file by default when you run the `tracetest server install` command and select the `Using Docker Compose` option. The configuration file is mounted to `/app/config.yaml` within the Tracetest Docker container. When Tracetest is run with a `docker compose -f tracetest/docker-compose.yaml  up -d` command, the server will use the contents of this file to connect to the Postgres database. All other configuration data is stored in the Postgres instance.
```

File: docs/docs/configuration/server.md [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+
+This is an example of a tracetest.yaml file:
```
Comment (adnanrahic) [inline docs/docs/configuration/server.md:7]:
```suggestion
This is an example of a `tracetest.yaml` file:
```

File: docs/docs/configuration/server.md [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+  port: 5432
+  dbname: postgres
+  params: sslmode=disable
+```
+
+Alternatively, we support setting a series of environment variables which can contain the connection information for the Postgres instance. If these environment variables are set, they will be used by the Tracetest server to connect to the database.
+
+The list of environment variables and example values is:
+- TRACETEST_POSTGRES_HOST - example: postgres
+- TRACETEST_POSTGRES_PORT - example: 5432
```
Comment (adnanrahic) [inline docs/docs/configuration/server.md:23]:
```suggestion
- `TRACETEST_POSTGRES_PORT: "5432"`
```

File: docs/docs/configuration/server.md [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+- TRACETEST_POSTGRES_DBNAME - example: postgres
+- TRACETEST_POSTGRES_USER - example: postgres
```
Comment (adnanrahic) [inline docs/docs/configuration/server.md:25]:
```suggestion
- `TRACETEST_POSTGRES_USER: "postgres"`
```

**Review by adnanrahic — COMMENTED** [review]
Added a few slight suggestions.

File: docs/docs/configuration/server.md [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+```
+
+Alternatively, we support setting a series of environment variables which can contain the connection information for the Postgres instance. If these environment variables are set, they will be used by the Tracetest server to connect to the database.
+
+The list of environment variables and example values is:
+- `TRACETEST_POSTGRES_HOST: "postgres"`
+- `TRACETEST_POSTGRES_PORT: "5432"`
+- `TRACETEST_POSTGRES_DBNAME: "postgres"`
+- `TRACETEST_POSTGRES_USER: "postgres"`
+- `TRACETEST_POSTGRES_PASSWORD: "postgres"`
```
Comment (jorgeepc) [inline docs/docs/configuration/server.md:26]: We also support `TRACETEST_POSTGRES_PARAMS`

**Review by jorgeepc — APPROVED** [review]

**Pull request merged**